src/Monolog/Handler/RotatingFileHandler.php

```diff
@@ -80,6 +85,10 @@ protected function write(array $record)
             $this->mustRotate = !file_exists($this->url);
         }
 
+        if (null === $this->mustSymlink) {
+            $this->mustSymlink = !is_link($this->filename);
+        }
```

Comment:
You are completely right, this was a mistake. I have updated/simplified my PR ; now the symlink is updated on each rotation.